### PR TITLE
fix: automatic resource sizing script & add kubelet sysconfig path for DEB systems

### LINKED ISSUE #1728: invalid kubelet configuration after kubelet restart when resource-sizing is enabled
## Environment

- **Make target:** `make build-qemu-ubuntu-2404`
- **Run using container image? (Y/N):** Y
- **Environment vars:** kubernetes_enable_automatic_resource_sizing=true

## What steps did you take and what happened?

- build a ubuntu image with `kubernetes_enable_automatic_resource_sizing=true`
- deploy an instance to test
- the first kubelet start works, `KUBELET_EXTRA_ARGS` variable is correclty defined in /run/kubelet/extra-args.env
- restart kubelet
- kubelet starts with a partial configuration
- `KUBELET_EXTRA_ARGS` is not correctly defined in `/run/kubelet/extra-args.env` 

## What did you expect to happen?

A valid behaviour between kubelet restarts.

## Relevant log output

<details>
<summary>Log Output</summary>

```bash
# first start
cat /run/kubelet/extra-args.env
KUBELET_EXTRA_ARGS=--pod-infra-container-image=registry.k8s.io/pause:3.9 --system-reserved=cpu=80m,memory=1390Mi

# on restart
systemctl restart kubelet
journalctl -u kubelet -e --no-pager
[...]
May 07 09:46:35 kaas-infra-dev-md-0-74prr-nss5x systemd[1]: Starting kubelet.service - kubelet: The Kubernetes Node Agent...
May 07 09:46:35 kaas-infra-dev-md-0-74prr-nss5x kubelet-resource-sizing.sh[7512]: /usr/libexec/kubernetes/kubelet-resource-sizing.sh: line 11: /etc/default/sysconfig: No such file or directory
May 07 09:46:35 kaas-infra-dev-md-0-74prr-nss5x kubelet-resource-sizing.sh[7519]: --pod-infra-container-image=registry.k8s.io/pause:3.9 --system-reserved=cpu=80m,memory=1390Mi
[...]

cat /run/kubelet/extra-args.env
--pod-infra-container-image=registry.k8s.io/pause:3.9 --system-reserved=cpu=80m,memory=1390Mi
```
</details>

## Anything else you would like to add?

https://github.com/kubernetes-sigs/image-builder/blob/f324366fffe17ca124685fd36b7ff9ee135cf6e6/images/capi/ansible/roles/kubernetes/files/usr/libexec/kubernetes/kubelet-resource-sizing.sh#L6

this source path is incorrect for deb distributions. (/etc/default/kubelet)

https://github.com/kubernetes-sigs/image-builder/blob/f324366fffe17ca124685fd36b7ff9ee135cf6e6/images/capi/ansible/roles/kubernetes/files/usr/libexec/kubernetes/kubelet-resource-sizing.sh#L114-L117

in this statement, the variable definition appended to /run/kubelet/extra-args.env is not correctly defined.

---

/kind bug

### PR DESCRIPTION
**Pull request by ffais**
## Change description
Fixes wrong variable declaration in /run/kubelet/extra-args.env when kubelet restart and --system-reserved is already set. 

I also added support to the kubelet sysconfig file for deb systems. Now DEB and RPM are supported, if we need to support other systems with different paths just add it to the variable "KUBELET_SYSCONFIG_FILES"

## Related issues

- Fixes #1728

### CODE REVIEW

File: images/capi/ansible/roles/kubernetes/files/usr/libexec/kubernetes/kubelet-resource-sizing.sh [hunk 1]
```diff
@@ -114,7 +117,7 @@ mkdir -p /run/kubelet
 # Check if system-reserved already exists
 if grep '.*--system-reserved' <<< "${KUBELET_EXTRA_ARGS}"; then
   # If system-reserved is already set by a previous run, replace old value with new one and write to /run/kubelet/extra-args.env
-  sed -E "s|--system-reserved=cpu=[0-9]+m,memory=[0-9]+Mi|--system-reserved=cpu=$(cpu_milicores_to_reserve)m,memory=$(memory_reservation_mebibytes)Mi|" <<< "${KUBELET_EXTRA_ARGS}" >/run/kubelet/extra-args.env
+  sed -i -E "s|--system-reserved=cpu=[0-9]+m,memory=[0-9]+Mi|--system-reserved=cpu=$(cpu_milicores_to_reserve)m,memory=$(memory_reservation_mebibytes)Mi|" /run/kubelet/extra-args.env
```
Comment (AverageMarcus) [inline images/capi/ansible/roles/kubernetes/files/usr/libexec/kubernetes/kubelet-resource-sizing.sh:120]: The `-i` flag isn't completely portable cross-platform. The built-in Mac version is slightly different to the GNU one 😞 

https://stackoverflow.com/a/12696224

Comment (ffais) [reply images/capi/ansible/roles/kubernetes/files/usr/libexec/kubernetes/kubelet-resource-sizing.sh:120]: Then it is better to remove the -i flag, what do you think about this solution?
```bash 
  system_reserved=$(sed -E "s|--system-reserved=cpu=[0-9]+m,memory=[0-9]+Mi|--system-reserved=cpu=$(cpu_milicores_to_reserve)m,memory=$(memory_reservation_mebibytes)Mi|" <<< "${KUBELET_EXTRA_ARGS}")
  echo "KUBELET_EXTRA_ARGS=${system_reserved} >/run/kubelet/extra-args.env"
```

Comment (AverageMarcus) [reply images/capi/ansible/roles/kubernetes/files/usr/libexec/kubernetes/kubelet-resource-sizing.sh:120]: Looks good to me if it works :)

Comment (ffais) [reply images/capi/ansible/roles/kubernetes/files/usr/libexec/kubernetes/kubelet-resource-sizing.sh]: https://github.com/kubernetes-sigs/image-builder/blob/973204628b7c531b3007ad546f590ae28698dc38/images/capi/ansible/roles/kubernetes/files/usr/libexec/kubernetes/kubelet-resource-sizing.sh#L118-L121

This part of the script is useful in the edge case where the KUBELET_SYSCONFIG file is not present and the KUBELET_EXTRA_ARGS variable has been configured by a previous execution of the script. I did some tests with Ubuntu 24.04 and everything seems to work fine.

**Review by AverageMarcus — COMMENTED** [review]
/ok-to-test
/lgtm
/assign @mboersma

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

Thanks @ffais!
